autoload/lsc/edit.vim

```diff
@@ -5,7 +5,8 @@ function! lsc#edit#findCodeActions(...) abort
     let ActionFilter = function("<SID>ActionMenu")
   endif
   call lsc#file#flushChanges()
-  let params = lsc#params#documentRange()
+  let l:usingRange = a:0 > 2 && (a:2 != 1 || a:3 != line('$'))
```

Comment:
Why do you need to check `a:2 != 1 || a:3 != line('$')`?

Reply:
This was my hack to know whether we were called with a range. If we're called like `:LSClientFindCodeActions` then the `-range=%` at the command definition will fill in `1,$` and we're identifying that case here. If we're called like `:'<,'>LSClientFindCodeActions` then it shouldn't match `1,$` and we'd know to actually look at the visual selection marks.

It does mean you can't actually do `:1,$LSClientFindCodeActions` and pass the entire doc in the range to the server...

Reply:
Should it be `&&` instead of `||` then?

Also sending the whole document range can be useful. Example: wrapping code in IIFE.
Maybe it is better to expose function with argument `isRange` and use it in keybindings instead of command which tries to guess visual mode?